src/resource.ts

```diff
@@ -0,0 +1,340 @@
+   * @returns The resource
+   *
+   * @since 1.0.0
+   *
+   * @public ✓ This method is part of the public API
+   */
+  public async create(prms: any, cb?: ResourceCallback): Promise<Model> {
+    const params = cloneDeep(prms);
+    const callback = typeof params === 'function' ? params : cb;
+    const query: any = {};
+    if (params !== null && typeof params === 'object') {
+      if (typeof params.include === 'string') {
+        query.include = params.include;
+        delete params.include;
+      }
+      if (typeof params.embed !== 'undefined' && Array.isArray(params.embed)) {
+        query.embed = params.embed.join(';');
+        delete params.embed;
+      }
+    }
+
+    try {
+      const response: AxiosResponse = await this.getClient().post(
+        `${this.getResourceUrl()}${qs.stringify(query, { addQueryPrefix: true })}`,
+        params,
+      );
+
+      // noinspection JSPotentiallyInvalidConstructorUsage
+      const model = new (this.constructor as any).model(response.data);
+
+      if (callback) {
+        return callback(null, model);
+      }
+      return model;
+    } catch (error) {
+      throw Resource.errorHandler(error.response, callback);
+    }
+  }
+
+  /**
+   * Get a resource by ID
+   *
+   * @params id - Resource ID
+   * @params params - Resource-specific parameters
+   * @params cb - (DEPRECATED SINCE 2.2.0) Optional callback function
+   *
+   * @returns {Promise<Model>}
+   *
+   * @since 1.0.0
+   */
+  public async get(id: string, prms?: any, cb?: ResourceCallback): Promise<Model> {
+    const params = cloneDeep(prms);
+    const callback = typeof params === 'function' ? params : cb;
+    const query: any = {};
+    if (params != null && typeof params === 'object') {
+      if (typeof params.include === 'string') {
+        query.include = params.include;
+        delete params.include;
+      }
+      if (typeof params.embed !== 'undefined' && Array.isArray(params.embed)) {
+        query.embed = params.embed.join(';');
+        delete params.embed;
+      }
+    }
+
+    try {
+      const response: AxiosResponse = await this.getClient().get(
+        `${this.getResourceUrl()}/${id}${qs.stringify(query, { addQueryPrefix: true })}`,
+      );
+
+      // noinspection JSPotentiallyInvalidConstructorUsage
+      const model = new (this.constructor as any).model(response.data);
+
+      if (callback) {
+        return callback(null, model);
+      }
+
+      return model;
+    } catch (error) {
+      throw Resource.errorHandler(error.response, callback);
+    }
+  }
+
+  /**
+   * List resources
+   *
+   * @param prms - Resource-specific parameters
+   * @param cb - (DEPRECATED SINCE 2.2.0) Optional callback function
+   *
+   * @returns Resource list
+   *
+   * @since 1.0.0
+   */
+  public async list(prms?: any, cb?: ResourceCallback): Promise<List<Model>> {
+    const params = cloneDeep(prms);
+    try {
+      const query: any = {};
+      if (params != null && typeof params === 'object') {
```

Comment:
Can you bring this entire class under tests? Especially the embeds / include parts.